server/neptune/gateway/event/push_event_handler.go

```diff
@@ -80,22 +84,35 @@ func (p *PushHandler) Handle(ctx context.Context, event Push) error {
 }
 
 func (p *PushHandler) handle(ctx context.Context, event Push) error {
-	options := client.StartWorkflowOptions{TaskQueue: workflows.DeployTaskQueue}
-
-	// TODO: clone and build project config
+	rootCfgs, err := p.RootConfigBuilder.BuildRootConfigs(ctx, event)
+	if err != nil {
+		return errors.Wrap(err, "generating roots")
+	}
+	for _, rootCfg := range rootCfgs {
+		ctx = context.WithValue(ctx, contextInternal.ProjectKey, rootCfg.Name)
+		run, err := p.startWorkflow(ctx, event, rootCfg)
+		if err != nil {
+			return errors.Wrap(err, "signalling workflow")
+		}
+
+		p.Logger.InfoContext(ctx, "Signaled workflow.", map[string]interface{}{
+			"workflow-id": run.GetID(), "run-id": run.GetRunID(),
+		})
+	}
+	return nil
+}
 
+func (p *PushHandler) startWorkflow(ctx context.Context, event Push, rootCfg *valid.MergedProjectCfg) (client.WorkflowRun, error) {
+	options := client.StartWorkflowOptions{TaskQueue: workflows.DeployTaskQueue}
 	run, err := p.TemporalClient.SignalWithStartWorkflow(
 		ctx,
-
-		// TODO: name should include root name as well
-		event.Repo.FullName,
+		fmt.Sprintf("%s||%s", event.Repo.FullName, rootCfg.Name),
```

Comment:
Open to suggestions on how to best structure the workflow id. We have to avoid using dashes, underscores, and periods as delimiters since GH allows those as valid repo chars.

Reply:
What you have probably works, just make sure it's not url encoded, i think with temporallite that breaks.